pkg/core/core.go

```diff
@@ -683,3 +683,258 @@ func GetThanosStoreService(plantDCore *windtunnelv1alpha1.PlantDCore) *corev1.Se
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"autoscaling"},
+				Resources: []string{"horizontalpodautoscalers"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"policy"},
+				Resources: []string{"poddisruptionbudgets"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"storage.k8s.io"},
+				Resources: []string{"storageclasses"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		},
+	}
+
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: config.GetString("opencost.clusterRole"),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     config.GetString("opencost.clusterRole"),
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      config.GetString("opencost.clusterRole"),
+				Namespace: plantDCore.Namespace,
+			},
+		},
+	}
+	return sa, clusterRole, clusterRoleBinding
+}
+func GetOpencostDeployment(plantDCore *windtunnelv1alpha1.PlantDCore) *appsv1.Deployment {
+	labels := config.GetStringMapString("opencost.deployment.labels")
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "opencost",
+			Namespace: plantDCore.Namespace,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.To(config.GetInt32("opencost.deployment.replicas")),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxSurge:       ptr.To(intstr.FromInt32(config.GetInt32("opencost.deployment.maxSurge"))),
+					MaxUnavailable: ptr.To(intstr.FromInt32(config.GetInt32("opencost.deployment.maxUnavailable"))),
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyAlways,
+					ServiceAccountName: "opencost",
+					Containers: []corev1.Container{
+						{
+							Name:  "opencost",
+							Image: config.GetString("opencost.deployment.image"),
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resourceQuantity("10m"),
+									corev1.ResourceMemory: resourceQuantity("55M"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resourceQuantity("999m"),
+									corev1.ResourceMemory: resourceQuantity("1G"),
+								},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "PROMETHEUS_SERVER_ENDPOINT", Value: config.GetString("database.prometheus.url")},
+								{Name: "CLOUD_PROVIDER_API_KEY", Value: config.GetString("opencost.deployment.cloudProviderAPIKey")},
+								{Name: "CLUSTER_ID", Value: "cluster-one"},
```

Comment:
Is the `CLUSTER_ID` fixed or not?

Reply:
Opencost gets data from prometheus. This is the default fallback in case prometheus metrics don't have CLUSTER_ID tag